plugins/node/opentelemetry-instrumentation-mysql/src/instrumentation.ts

```diff
@@ -50,9 +52,27 @@ export class MySQLInstrumentation extends InstrumentationBase<
   static readonly COMMON_ATTRIBUTES = {
     [SemanticAttributes.DB_SYSTEM]: DbSystemValues.MYSQL,
   };
+  private _connectionsUsage!: UpDownCounter;
```

Comment:
The `!` assertion could be removed if the metric creation was done in the constructor. Is there a reason it is split into a separate `setMetricInstruments` call?

Reply:
regarding the call to `setMetricInstruments` - we call `createUpDownCounter` to create a counter that is related to a `meterProvider`. If we call it in the constructor, and the user after that will call `setMeterProvider` (which he can..), it will override the `meterProvider`.
This is why I put it in a separate function called `_setMetricInstruments`, so it will be called in any case.

Reply:
See also https://github.com/open-telemetry/opentelemetry-js/pull/3267. The meter provider may be set multiple times. The metric instruments need to be re-created once that happens.

Reply:
@legendecas Should I wait until this issue (#3267) will be merged? When will it happen?

Reply:
I think it is ok to proceed without that patch. It may need to wait for the next release cycle.

Reply:
ok. thanks. @dyladan is it ok to keep the `!` assertion? 
I can't initialize it in the constructor because the initialization is internal in `_setMetricInstruments`.